### PR TITLE
[test] Add lacp fallback test cases

### PR DESCRIPTION
**Pull request by hzheng5**
Signed-off-by: Haiyang Zheng <haiyang.z@alibaba-inc.com>

### Description of PR

Summary:
1. Run lag_fallback test cases if fallback is configure on lags with only 1 member port
2. Run lag_minlink test cases only if min_ports is configured on lags with more than 2 member ports

**Please note min_port and fallback configure are mutual exclusive for each lag.**

### Type of change

- [] Bug fix
- [] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

#### How did you do it?

The lag_fallback test cases flap the link from VM side while keeping the physical link up, this is to simulate the situation where the remote end stops sending LACP DU. If fallback is enabled, the port should still be selected, and LAG should be kept up. Otherwise, the member port should be unselected, and LAG should be brought down.

#### How did you verify/test it?
ansible-playbook test_sonic.yml -i {INVENTORY} --limit {DUT_NAME} -e testcase_name=lag_2 -e testbed_name={TESTBED_NAME}

#### Any platform specific information?

no
#### Supported testbed topology if it's a new test case?

t0 topo

### Documentation 

https://github.com/Azure/SONiC/blob/master/doc/lag/LACP%20Fallback%20Feature%20for%20SONiC_v0.5.md

### CODE REVIEW

File: ansible/library/minigraph_facts.py [hunk 1]
```diff
@@ -212,7 +212,10 @@ def parse_dpg(dpg, hname):
             for i, member in enumerate(pcmbr_list):
                 pcmbr_list[i] = port_alias_to_name_map[member]
                 ports[port_alias_to_name_map[member]] = {'name': port_alias_to_name_map[member], 'alias': member}
-            pcs[pcintfname] = {'name': pcintfname, 'members': pcmbr_list}
+            if pcintf.find(str(QName(ns, "Fallback"))) != None:
```
[the remark this answers is not in the document]
Comment (hzheng5) [reply ansible/library/minigraph_facts.py]: Thanks @pavel-shirshov for reviewing the changes.

Has updated the code as suggested.

**Comment by pavel-shirshov** [comment]
I'll double review it in a day. Sorry for delay. I want to be sure it is safe with old tests.

**Comment by hzheng5** [comment]
> I'll double review it in a day. Sorry for delay. I want to be sure it is safe with old tests.

sure. no problem. thanks~

File: ansible/roles/test/tasks/lag_2.yml [hunk 2]
```diff
@@ -72,9 +72,14 @@
 - name: test each lag interface minimum links and rate
   include: single_lag_test.yml
   with_items: lag_facts.names
-  when: test_minlink|bool == true
+  when: lag_facts.lags[item]['po_config']['runner']['min_ports'] is defined
```
Comment (pavel-shirshov) [inline ansible/roles/test/tasks/lag_2.yml]: I think you should keep the test_minlink here. The test could be disabled when test_minlink is equal to false.

File: ansible/roles/test/tasks/lag_2.yml [hunk 3]
```diff
@@ -72,9 +72,14 @@
...
 
 - name: test each lag interface LACP DU rate
   include: single_lag_lacp_rate_test.yml
   with_items: lag_facts.names
-  when: test_rate|bool == true
\ No newline at end of file
+  when: lag_facts.lags[item]['po_config']['runner']['min_ports'] is defined
```
Comment (pavel-shirshov) [inline ansible/roles/test/tasks/lag_2.yml]: The same here for test_rate.
Please fix it and I'll merge

File: ansible/roles/test/tasks/lag_2.yml (hunk 2 above)
Comment (hzheng5) [reply ansible/roles/test/tasks/lag_2.yml]: OK. got your point.  have reverted this change caused by merge. Thanks

File: ansible/roles/test/tasks/lag_2.yml (hunk 3 above)
Comment (hzheng5) [reply ansible/roles/test/tasks/lag_2.yml]: done. thanks!

File: ansible/roles/test/tasks/lag_2.yml [hunk 4]
```diff
@@ -72,12 +72,12 @@
 - name: test each lag interface minimum links and rate
   include: single_lag_test.yml
   with_items: lag_facts.names
-  when: lag_facts.lags[item]['po_config']['runner']['min_ports'] is defined
+  when: test_minlink|bool == true
```
Comment (pavel-shirshov) [inline ansible/roles/test/tasks/lag_2.yml]: `lag_facts.lags[item]['po_config']['runner']['min_ports'] is defined and test_minlink|bool == true`

File: ansible/roles/test/tasks/lag_2.yml [hunk 5]
```diff
@@ -72,12 +72,12 @@
...
 
 - name: test each lag interface LACP DU rate
   include: single_lag_lacp_rate_test.yml
   with_items: lag_facts.names
-  when: lag_facts.lags[item]['po_config']['runner']['min_ports'] is defined
+  when: test_rate|bool == true
```
Comment (pavel-shirshov) [inline ansible/roles/test/tasks/lag_2.yml]: `lag_facts.lags[item]['po_config']['runner']['min_ports'] is defined and test_rate|bool == true`

Comment (hzheng5) [reply ansible/roles/test/tasks/lag_2.yml]: done

**Comment by chenkelly** [comment]
Hi @hzheng5 
Please guide us how to deploy minigraph and test  LAG fallback.
When we merge code and use the following command to deploy, LAG fallback configuration does not apply to specified DUT. 
Could you give us some advice about the question? Thanks very much.
ansible-playbook -i lab config_sonic_basedon_testbed.yml -l {DUT_NAME} -e testbed_name={TESTBED_NAME} -e deploy=true -e save=true

**Pull request merged**